<launch>
  <!-- Run the depthimage_to_laserscan node -->
  <node pkg="depthimage_to_laserscan" type="depthimage_to_laserscan" name="depthimage_to_laserscan" output="screen">
    <!--input depth img-->
     <remap from="image" to="/camera/depth/image_rect_raw"/>
    <!--out the lidar rostopic-->
    <remap from="scan" to="/scan" /> 
    <!--The number of pixel rows to use to generate the laserscan.-->
    <param name="scan_height" value="20"/>
    <param name="scan_height_min" value="700"/>
    <param name="scan_height_max" value="900"/>
    <!--Time between scans (seconds). Typically, 1.0/frame_rate. -->
　　<param name="scan_time" value="0.03"/>
    <!--Minimum ranges to return in meters. Ranges less than this will be output as -Inf.-->
    <param name="range_min" value="0.4"/>
    <!--Maximum ranges to return in meters. Ranges greater than this will be output as +Inf.-->
    <param name="range_max" value="8.0"/>

    <param name="output_frame_id" value="camera_depth_frame"/>
    
    <param name="show_mask" value="true"/>
  </node>
	
  <node pkg="rviz" type="rviz" name="rviz" required="true"
          args="-d $(find depthimage_to_laserscan)/rviz/check_range.rviz"/>
          
  <node pkg="laser_filters" type="scan_to_scan_filter_chain" name="laser_filter">
      <rosparam command="load" file="$(find depthimage_to_laserscan)/config/lidar_config.yaml"/>
    <remap from="base_scan" to="scan"/>
  </node>
  
</launch>
